<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1359</width>
    <height>686</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="imgLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>491</width>
      <height>521</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="imgLabel2">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>20</y>
      <width>491</width>
      <height>521</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QDial" name="dial">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>10</y>
      <width>81</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>30</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Rotation</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>90</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>5</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
    <property name="sliderPosition">
     <number>5</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>90</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Gamma</string>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>10</y>
      <width>20</width>
      <height>631</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSlider" name="size_Img">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>60</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
    <property name="sliderPosition">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>60</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Scaling</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>120</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Gaussian</string>
    </property>
   </widget>
   <widget class="QSlider" name="gaussian_QSlider">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>120</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>4</number>
    </property>
    <property name="pageStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="erosion">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>170</y>
      <width>201</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5</number>
    </property>
    <property name="maximum">
     <number>5</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>150</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Dilate_Erosion</string>
    </property>
   </widget>
   <widget class="QLabel" name="log">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>200</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Log</string>
    </property>
   </widget>
   <widget class="QSlider" name="Qlog">
    <property name="geometry">
     <rect>
      <x>1170</x>
      <y>200</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>4</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>240</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>310</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Min</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>360</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Max</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="canny">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>270</y>
      <width>171</width>
      <height>22</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>Canny Edge Detector</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QSlider" name="canny_min">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>310</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="canny_max">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>360</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>400</number>
    </property>
    <property name="value">
     <number>200</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>1170</x>
      <y>460</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
    <property name="text">
     <string>RESET</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>540</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>24</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Original</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>540</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>24</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Changed</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1359</width>
     <height>36</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionPrint"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionSmall"/>
    <addaction name="actionBig"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_6">
    <property name="title">
     <string>Chapter 2</string>
    </property>
    <addaction name="actionRotation"/>
    <addaction name="actionAffine"/>
    <addaction name="actionTranslation"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_9">
    <property name="title">
     <string>Chapter 3</string>
    </property>
    <addaction name="actioAnhXam"/>
    <addaction name="actionNegative"/>
    <addaction name="actionLog"/>
    <addaction name="actionHistogram"/>
    <addaction name="actionGamma"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About US</string>
    </property>
    <addaction name="actionQt"/>
    <addaction name="actionAuthor"/>
   </widget>
   <widget class="QMenu" name="menuType_Noise">
    <property name="title">
     <string>Image Restoration
</string>
    </property>
    <widget class="QMenu" name="menuType_Here">
     <property name="title">
      <string>Image Restoration 1
</string>
     </property>
     <addaction name="actionMedian_Filtering"/>
     <addaction name="actionAdaptive_Median_Filtering"/>
     <addaction name="actionAdaptive_Wiener_Filtering"/>
    </widget>
    <widget class="QMenu" name="menuImage_Restoration_2">
     <property name="title">
      <string>Image Restoration 2</string>
     </property>
     <addaction name="actionInverse_Filter"/>
    </widget>
    <widget class="QMenu" name="menuImage_Restoration">
     <property name="title">
      <string>Image Restoration </string>
     </property>
     <widget class="QMenu" name="menuType_Noise_3">
      <property name="title">
       <string>Type Noise</string>
      </property>
      <addaction name="actionGaussian"/>
      <addaction name="actionRayleigh"/>
      <addaction name="actionErlang"/>
      <addaction name="actionUniform"/>
      <addaction name="actionImpluse"/>
     </widget>
     <addaction name="menuType_Noise_3"/>
     <addaction name="separator"/>
     <addaction name="actionHistogram_PDF"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="menuImage_Restoration"/>
    <addaction name="menuType_Here"/>
    <addaction name="menuImage_Restoration_2"/>
   </widget>
   <widget class="QMenu" name="menuSimple_Edge_Detection_2">
    <property name="title">
     <string>Simple Edge Detection</string>
    </property>
    <addaction name="actionSHT"/>
   </widget>
   <widget class="QMenu" name="menuSmoothing">
    <property name="title">
     <string>Smoothing</string>
    </property>
    <addaction name="actionBlur"/>
    <addaction name="actionBox_Filter"/>
    <addaction name="actionMedian_Filter"/>
    <addaction name="actionBilateral_Filter"/>
    <addaction name="actionGaussian_Filter"/>
   </widget>
   <widget class="QMenu" name="menuFilter">
    <property name="title">
     <string>Filter</string>
    </property>
    <addaction name="actionMedian_threshold_2"/>
    <addaction name="actionDirectional_Filtering_2"/>
    <addaction name="actionDirectional_Filtering_3"/>
    <addaction name="actionDirectional_Filtering_4"/>
    <addaction name="action_Butterworth"/>
    <addaction name="action_Notch_filter"/>
   </widget>
   <widget class="QMenu" name="menuCartooning_of_an_Image">
    <property name="title">
     <string>Cartooning of an Image</string>
    </property>
    <addaction name="actionCartoon"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCh_ng_6"/>
   <addaction name="menuCh_ng_9"/>
   <addaction name="menuSimple_Edge_Detection_2"/>
   <addaction name="menuCartooning_of_an_Image"/>
   <addaction name="menuSmoothing"/>
   <addaction name="menuFilter"/>
   <addaction name="menuType_Noise"/>
   <addaction name="menuView"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionPrint"/>
   <addaction name="separator"/>
   <addaction name="actionQuit"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>open.png</normaloff>open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>save.png</normaloff>save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset>
     <normaloff>print.png</normaloff>print.png</iconset>
   </property>
   <property name="text">
    <string>Print</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset>
     <normaloff>Quit.png</normaloff>Quit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actioAnhXam">
   <property name="text">
    <string>Graycsale</string>
   </property>
  </action>
  <action name="actionCanny">
   <property name="text">
    <string>Canny</string>
   </property>
  </action>
  <action name="actionNegative">
   <property name="text">
    <string>Negative</string>
   </property>
  </action>
  <action name="actionHistogram">
   <property name="text">
    <string>Histogram Equal</string>
   </property>
  </action>
  <action name="actionLog">
   <property name="text">
    <string>Log </string>
   </property>
  </action>
  <action name="actionGaussan">
   <property name="text">
    <string>Gaussan</string>
   </property>
  </action>
  <action name="actionHigh_Boost">
   <property name="text">
    <string>High Boost</string>
   </property>
  </action>
  <action name="actionLaplacian">
   <property name="text">
    <string>Laplacian</string>
   </property>
  </action>
  <action name="actionFilter_Average">
   <property name="text">
    <string>Median</string>
   </property>
  </action>
  <action name="actionUnsharp">
   <property name="text">
    <string>Unsharp</string>
   </property>
  </action>
  <action name="actionCh_ng_5">
   <property name="text">
    <string>Chương 5</string>
   </property>
  </action>
  <action name="actionTanSo">
   <property name="text">
    <string>Ảnh Miền Tần Số</string>
   </property>
  </action>
  <action name="actionIdeal_LPF">
   <property name="text">
    <string>Ideal LPF</string>
   </property>
  </action>
  <action name="actionButter_LPF">
   <property name="text">
    <string>Butterworth LPF</string>
   </property>
  </action>
  <action name="actionGaussian_LPF">
   <property name="text">
    <string>Gaussian LPF</string>
   </property>
  </action>
  <action name="actionIdeal_HPF">
   <property name="text">
    <string>Ideal HPF</string>
   </property>
  </action>
  <action name="actionButterworth_HPF">
   <property name="text">
    <string>Butterworth HPF</string>
   </property>
  </action>
  <action name="actionGaussian_HPF">
   <property name="text">
    <string>Gaussian HPF</string>
   </property>
  </action>
  <action name="actiondilate">
   <property name="text">
    <string>dilate</string>
   </property>
  </action>
  <action name="actionErode">
   <property name="text">
    <string>Erode</string>
   </property>
  </action>
  <action name="actionOpen_2">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionHit_miss">
   <property name="text">
    <string>Hit-miss</string>
   </property>
  </action>
  <action name="actionDilate">
   <property name="text">
    <string>Dilate</string>
   </property>
  </action>
  <action name="actionMorboundary">
   <property name="text">
    <string>Morboundary</string>
   </property>
  </action>
  <action name="actionGradient">
   <property name="text">
    <string>Gradient</string>
   </property>
  </action>
  <action name="actionConvex">
   <property name="text">
    <string>Convex</string>
   </property>
  </action>
  <action name="actionx_direcction_Sobel">
   <property name="text">
    <string>Sobel X</string>
   </property>
  </action>
  <action name="actiony_direction_Sobel">
   <property name="text">
    <string>Sobel Y</string>
   </property>
  </action>
  <action name="actionLaplacian_2">
   <property name="text">
    <string>Sobel Laplacian</string>
   </property>
  </action>
  <action name="actionLaplacian_of_Gaussian">
   <property name="text">
    <string>Laplacian of Gaussian</string>
   </property>
  </action>
  <action name="actionHough">
   <property name="text">
    <string>Hough</string>
   </property>
  </action>
  <action name="actionSmall">
   <property name="text">
    <string>Zoom in</string>
   </property>
  </action>
  <action name="actionRotation">
   <property name="text">
    <string>Rotation</string>
   </property>
  </action>
  <action name="actionAffine">
   <property name="text">
    <string>Shearing</string>
   </property>
  </action>
  <action name="actionGamma">
   <property name="text">
    <string>Gamma</string>
   </property>
  </action>
  <action name="actionBig">
   <property name="text">
    <string>Zoom out</string>
   </property>
  </action>
  <action name="actionQt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionAuthor">
   <property name="text">
    <string>Author</string>
   </property>
  </action>
  <action name="actionTranslation">
   <property name="text">
    <string>Translation</string>
   </property>
  </action>
  <action name="actionGaussian">
   <property name="text">
    <string>Gaussian</string>
   </property>
  </action>
  <action name="actionRayleigh">
   <property name="text">
    <string>Rayleigh</string>
   </property>
  </action>
  <action name="actionImpluse">
   <property name="text">
    <string>Impluse</string>
   </property>
  </action>
  <action name="actionUniform">
   <property name="text">
    <string>Uniform</string>
   </property>
  </action>
  <action name="actionErlang">
   <property name="text">
    <string>Erlang</string>
   </property>
  </action>
  <action name="actionHistogram_PDF">
   <property name="text">
    <string>Histogram PDF</string>
   </property>
  </action>
  <action name="actionDetecion">
   <property name="text">
    <string>HoughLines</string>
   </property>
  </action>
  <action name="actionHoughLines">
   <property name="text">
    <string>HoughLines</string>
   </property>
  </action>
  <action name="actionSHT">
   <property name="text">
    <string> Standard Hough Transform</string>
   </property>
  </action>
  <action name="actionMedian_Filtering">
   <property name="text">
    <string>Median Filtering</string>
   </property>
  </action>
  <action name="actionAdaptive_Wiener_Filtering">
   <property name="text">
    <string>Adaptive Wiener Filtering</string>
   </property>
  </action>
  <action name="actionAdaptive_Median_Filtering">
   <property name="text">
    <string>Adaptive Median Filtering</string>
   </property>
  </action>
  <action name="actionInverse_Filter">
   <property name="text">
    <string>Inverse Filter </string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
  <action name="actionBox_Filter">
   <property name="text">
    <string>Box Filter</string>
   </property>
  </action>
  <action name="actionMedian_Filter">
   <property name="text">
    <string>Median Filter</string>
   </property>
  </action>
  <action name="actionBilateral_Filter">
   <property name="text">
    <string>Bilateral Filter</string>
   </property>
  </action>
  <action name="actionGaussian_Filter">
   <property name="text">
    <string>Gaussian Filter</string>
   </property>
  </action>
  <action name="actionDirectional_Filtering">
   <property name="text">
    <string>Directional Filtering</string>
   </property>
  </action>
  <action name="actionMedian_threshold">
   <property name="text">
    <string>Median threshold</string>
   </property>
  </action>
  <action name="actionMedian_threshold_2">
   <property name="text">
    <string>Median threshold</string>
   </property>
  </action>
  <action name="actionDirectional_Filtering_2">
   <property name="text">
    <string>Directional Filtering</string>
   </property>
  </action>
  <action name="action_Butterworth">
   <property name="text">
    <string>Butterworth Filter</string>
   </property>
  </action>
  <action name="action_Notch_filter">
   <property name="text">
    <string>Notch Filter</string>
   </property>
  </action>
  <action name="actionCartoon">
   <property name="text">
    <string>Cartoon</string>
   </property>
  </action>
  <action name="actionDirectional_Filtering_3">
   <property name="text">
    <string>Directional Filtering 2</string>
   </property>
  </action>
  <action name="actionDirectional_Filtering_4">
   <property name="text">
    <string>Directional Filtering 3</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
